nemo/lightning/run/plugins.py

```diff
@@ -241,3 +242,64 @@ def setup(self, task: run.Partial | run.Script, executor: run.Executor):
             if isinstance(executor, run.SlurmExecutor):
                 assert task.trainer.num_nodes == executor.nodes
                 assert task.trainer.devices == executor.nproc_per_node()
+
+
+@dataclass(kw_only=True)
+class PerfEnvPlugin(run.Plugin):
+    """
+    A plugin for setting up performance optimized environments.
+
+    Attributes:
+        enable_vboost (bool): Whether to steer more power towards tensor cores via
+            `sudo nvidia-smi boost-slider --vboost 1`. May not work on all systems.
+    """
+
+    enable_vboost: bool = False
+
+    def get_vboost_srun_cmd(self, nodes, job_dir):
+        import shlex
+
+        vboost_cmd = " ".join(
+            [
+                "\n# Command 0: enable vboost\n\n",
+                "srun",
+                f"--ntasks={nodes}",
+                "--output",
+                os.path.join(job_dir, "vboost.out"),
+                "--error",
+                os.path.join(job_dir, "vboost.err"),
+                "bash -c ",
+                shlex.quote("sudo nvidia-smi boost-slider --vboost 1"),
+            ],
+        )
+
+        return vboost_cmd
+
+    def setup(self, task: run.Partial | run.Script, executor: run.Executor):
+
+        if task.trainer.strategy.__fn_or_cls__ == MegatronStrategy:
+            # Force program order kernel launch for TP, CP overlap
+            tp_size = task.trainer.strategy.tensor_model_parallel_size
+            cp_size = task.trainer.strategy.context_parallel_size
+            if tp_size > 1 and cp_size > 1:
+                executor.env_vars["CUDA_DEVICE_MAX_CONNECTIONS"] = 1
+
+            # Set LayerNorm SM margin when using P2P communication overlap to support the overlap with LayerNorm kernel
+            vpp = task.trainer.strategy.virtual_pipeline_model_parallel_size
+            pp_size = task.trainer.strategy.pipeline_model_parallel_size
+            if pp_size > 1 and vpp is not None and vpp > 1:
```

Comment:
Don't we need to set this true even at PP=1 because we have DP comm overlap with LN here as well?

Reply:
Okay, Ive put this behind a knob which defaults to true.